src/extrema.jl

```diff
@@ -0,0 +1,166 @@
+                  rthresh::Real=1//1000) where {T<:Union{AbstractGray,Real},N}
+    if edges isa Bool
+        edges = (edges, ntuple(d->edges,Val(N))...)
+    end
+    sigmas = sort(σscales)
+    img_LoG = multiLoG(img, sigmas, σshape)
+    maxima = findlocalmaxima(img_LoG; edges=edges)
+    # The "density" should not be much smaller than 1/volume of the Gaussian
+    if !iszero(rthresh)
+        athresh = rthresh./(sigmas.^N .* prod(σshape))
+        imgmax = maximum(abs, img)
+        [BlobLoG(CartesianIndex(tail(x.I)), sigmas[x[1]].*σshape, img_LoG[x]) for x in maxima if img_LoG[x] > athresh[x[1]]*imgmax]
+    else
+        [BlobLoG(CartesianIndex(tail(x.I)), sigmas[x[1]].*σshape, img_LoG[x]) for x in maxima]
+    end
+end
+
+function multiLoG(img::AbstractArray{T,N}, sigmas, σshape) where {T,N}
+    issorted(sigmas) || error("sigmas must be sorted")
+    img_LoG = similar(img, float(eltype(T)), (Base.OneTo(length(sigmas)), axes(img)...))
+    colons = ntuple(d->Colon(), Val(N))
+    @inbounds for (isigma, σ) in enumerate(sigmas)
+        LoG_slice = @view img_LoG[isigma, colons...]
+        imfilter!(LoG_slice, img, Kernel.LoG(ntuple(i->σ*σshape[i], Val(N))), "reflect")
+        LoG_slice .*= -σ
+    end
+    return img_LoG
+end
+
+default_window(img) = (cs = coords_spatial(img); ntuple(d -> d ∈ cs ? 3 : 1, ndims(img)))
+
+"""
+    findlocalmaxima(img; window=default_window(img), edges=true) -> Vector{CartesianIndex}
+
+Returns the coordinates of elements whose value is larger than all of
+their immediate neighbors.  `edges` is a boolean specifying whether to include the
+first and last elements of each dimension, or a tuple-of-Bool
+specifying edge behavior for each dimension separately.
+
+The `default_window` is 3 for each spatial dimension of `img`, and 1 otherwise, implying
+that maxima are detected over nearest-neighbors in each spatial "slice" by default.
+"""
+findlocalmaxima(img::AbstractArray; window=default_window(img), edges=true) =
+    findlocalextrema(>, img, window, edges)
+
+"""
+    findlocalminima(img; window=default_window(img), edges=true) -> Vector{CartesianIndex}
+
+Like [`findlocalmaxima`](@ref), but returns the coordinates of the smallest elements.
+"""
+findlocalminima(img::AbstractArray; window=default_window(img), edges=true) =
+    findlocalextrema(<, img, window, edges)
+
+
+findlocalextrema(f, img::AbstractArray{T,N}, window, edges::Bool) where {T,N} = findlocalextrema(f, img, window, ntuple(d->edges,Val(N)))
+
+function findlocalextrema(f::F, img::AbstractArray{T,N}, window::Dims{N}, edges::NTuple{N,Bool}) where {F,T<:Union{Gray,Number},N}
+    extrema = Vector{CartesianIndex{N}}(undef, 0)
+    Iedge = CartesianIndex(map(!, edges))
+    R0 = CartesianIndices(img)
+    R = clippedinds(R0, Iedge)
+    halfwindow = CartesianIndex(map(x -> x >> 1, window))
+    Rinterior = clippedinds(R0, halfwindow)
+    Rwindow = _colon(-halfwindow, halfwindow)
+    z = zero(halfwindow)
+    for i in R
```

Comment:
It looks valid for me to skip the default bounds checking here:

```suggestion
    @inbounds for i in R
```